js-core/homeworks/homework-4/src/main.js

```diff
@@ -0,0 +1,206 @@
+/*
+* Напишите функцию которая будет принимать 1 аргумент и в зависимости от типа аргумента
+*
+* Если тип аргумента число или объект -> возвращать true
+* Если тип аргумента функция -> возвращать false
+* Если тип аргумента Строка и длина этой строки не равна 10 -> возвращать "длина вашей строки: <длина строки>
+*                            Если длина 10 -> 'you win'
+* */
+  
+
+function returnType(arg){
+  if(typeof arg == 'number' || typeof arg == 'object' ){
+    return true
+  } else if (typeof arg == 'function'){
+    return false
+  } else if(typeof arg == 'string'){
+       if(arg.length == 10){
+          return 'you win'
+      }
+      return `Длинна вашей строки: ${arg.length}`
+  }
+}
+console.log(returnType(3));
+console.log(returnType({}));
+console.log(returnType(function(){}));  
+console.log(returnType('helloworld'));
+console.log(returnType('hello world'));
+
+
+
+
+
+// Задача в классе
+
+
+let webStore = {
+        category: {
+            notebook: ['Lenova', 'Asus', 'Xiaomi'],
+            mobile:['Nokia', 'Sony', 'Apple']
+        }
+    };
+
+    let category = webStore.category;
+
+    for(let key in category){
+        
+        console.log('Категория:', key);
+
+       for(let i = 0; i < category[key].length; i++){
```

Comment:
I think you have to define category[key] as a variable, that variable used widely and what does it mean not really clear.

For example, take a look on line 51, what does category[key][i] refer to ? what abstraction inside?